src/components/Search/SearchList.tsx

```diff
@@ -126,51 +126,15 @@ function SearchList(
     const {isSmallScreenWidth} = useResponsiveLayout();
 
     const [isModalVisible, setIsModalVisible] = useState(false);
-    const {selectionMode} = useMobileSelectionMode();
+    const selectionMode = useMobileSelectionMode();
     const [longPressedItem, setLongPressedItem] = useState<SearchListItem>();
-    // Check if selection should be on when the modal is opened
-    const wasSelectionOnRef = useRef(false);
-    // Keep track of the number of selected items to determine if we should turn off selection mode
-    const selectionRef = useRef(0);
 
     const [policies] = useOnyx(ONYXKEYS.COLLECTION.POLICY, {
         canBeMissing: true,
     });
 
     const [allReports] = useOnyx(ONYXKEYS.COLLECTION.REPORT, {canBeMissing: false});
 
-    useEffect(() => {
-        selectionRef.current = selectedItemsLength;
-
-        if (!isSmallScreenWidth) {
-            if (selectedItemsLength === 0) {
-                turnOffMobileSelectionMode();
-            }
-            return;
-        }
-        if (!isFocused) {
-            return;
-        }
-        if (!wasSelectionOnRef.current && selectedItemsLength > 0) {
-            wasSelectionOnRef.current = true;
-        }
-        if (selectedItemsLength > 0 && !selectionMode?.isEnabled) {
-            turnOnMobileSelectionMode();
-        } else if (selectedItemsLength === 0 && selectionMode?.isEnabled && !wasSelectionOnRef.current) {
-            turnOffMobileSelectionMode();
-        }
-    }, [selectionMode, isSmallScreenWidth, isFocused, selectedItemsLength]);
-
-    useEffect(
-        () => () => {
-            if (selectionRef.current !== 0) {
-                return;
-            }
-            turnOffMobileSelectionMode();
-        },
-        [],
-    );
-
```

Comment:
amazing 👏 
Can you tell me in short what part of code is responsible for disabling selection mode now in these cases?